File: Introduccion a Poo/Ejercicio 10/Java/Avion.java
public class Avion {
    private int pasajeros;
    private int combustible;
    private String horaVuelo;
    public Avion() {
        pasajeros = 0;
        combustible = 100000;
        horaVuelo = "10:00";
    }
    public void subirPasajeros(int cantidad) {
        pasajeros += cantidad;
    }
    public void verificarCombustible(int distancia) {
        int necesario = distancia * 12;

        if (necesario > combustible) {
            System.out.println("No hay suficiente combustible");
        } else {
            combustible -= necesario;
        }
    }
    public void retrasarVuelo() {
        horaVuelo = "13:00";
        System.out.println("El vuelo fue retrasado 3 horas");
    }
}

File: Introduccion a Poo/Ejercicio 10/Python/Avion.py
class Avion:
    def __init__(self):
        self.pasajeros = 0
        self.combustible = 100000
        self.hrVuelo = "10:00"
    def subir(self, cantidad):
        self.pasajeros = self.pasajeros + cantidad
    def conbustible(self, distancia):
        necesario = distancia * 12
        if necesario > self.combustible:
            print("No hay suficiente combustible")
        else:
            self.combustible = self.combustible - necesario
    def retraso(self):
        print("Vuelo retrasado 3 horas")
        self.hrVuelo = "13:00"
a = Avion()
a.subir(100)
a.conbustible(500)
a.retraso()
print("Nueva hora de vuelo:", a.hrVuelo)
